sml/asyncio.py

```diff
@@ -126,10 +131,23 @@ async def connect(self, loop=None):
         await self._reconnect(delay=0)
 
     async def _disconnect(self):
+        if self._watchdog and not self._watchdog.done():
+            self._watchdog.cancel()
         if self._transport:
             self._transport.abort()
             self._transport = None
 
+    async def _timeout(self):
+        timeout_delay = 120
```

Comment:
This value should probably be configurable.

Reply:
I have introduced a preconfigured variable within `__init__` which contains a time to wait. If this is 0 then the watchdog will not be started.